gnosis/safe/account_abstraction/safe_operation.py

```diff
@@ -87,11 +88,11 @@ def get_domain_separator(
             )
         return _domain_separator_cache[key]
 
-    def get_safe_operation_hash(
+    def get_safe_operation_hash_preimage(
```

Comment:
Could be cached_property and avoid to check if exists the pre-image?

Reply:
Dataclasses are not hashable by default, so cache decorators cannot be used. Also a property cannot be used as it receives parameters

Reply:
Maybe we should use namedtuple for this case https://stackoverflow.com/questions/51671699/data-classes-vs-typing-namedtuple-primary-use-cases

Reply:
> Dataclasses are not hashable by default, so cache decorators cannot be used. Also a property cannot be used as it receives parameters

Got it.
My concern is if you create an instance of `UserOperation`, maybe you are not aware that the `useroperation_hash` is `None` until call the get function. 
Maybe we can add `chain` and `module` as atributes of `UserOperation` like  `DOMAIN_HASH_SEPARATOR` is and remove `safe_operation_hash` and `safe_operation_hash_preimage` as parameters and create the property functions.

Reply:
Chain and module are not available when retrieving the UserOperation, that's why I don't want to force to define then as properties as they are not mandatory for working with the UserOperation and only needed when calculating the hash